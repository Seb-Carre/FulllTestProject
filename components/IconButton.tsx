import {ImageSourcePropType, Pressable, StyleSheet} from 'react-native';

import {Icon} from './Icon';
import React from 'react';

type IconButtonParams = {
  iconName: ImageSourcePropType;
  onPress: () => void;
  testID?: string;
};

/**
 * Pressable button with a Icon inside !
 */
export default function IconButton({
  iconName,
  onPress,
  testID,
}: IconButtonParams) {
  return (
    <Pressable testID={testID} onPress={onPress}>
      <Icon name={iconName} style={{...styles.icon}} />
    </Pressable>
  );
}

const styles = StyleSheet.create({
  icon: {
    height: 20,
    width: 20,
  },
});
