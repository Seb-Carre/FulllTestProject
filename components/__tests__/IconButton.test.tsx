import {fireEvent, render} from '@testing-library/react-native';

import IconButton from '../IconButton';
import React from 'react';

const checkMark = '../../assets/images/check-mark.png';
it('render the icon button as snapShot expected', () => {
  const {toJSON} = render(
    <IconButton iconName={require(checkMark)} onPress={() => {}} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('renders the icon button and click', () => {
  // Using a mock function to check if our function is called
  const mockFunction = jest.fn();
  const {getByTestId} = render(
    <IconButton
      testID="iconButton"
      iconName={require(checkMark)}
      onPress={mockFunction}
    />,
  );
  const button = getByTestId('iconButton');

  // Making a press event on our IconButton
  fireEvent.press(button);
  expect(mockFunction).toBeCalled();
});
