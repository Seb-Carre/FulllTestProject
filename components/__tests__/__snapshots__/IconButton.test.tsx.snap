// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the icon button as snapShot expected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/check-mark.png",
      }
    }
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
